### PR TITLE
Improve various docs on contributing and adds PR template

### PR DESCRIPTION
**Pull request by marcvanzee**
This PR tries to make it easier for OS users to submit high quality code to Flax:

- Adds guidelines for how to write code documentation based on Sphinx (`docs/README.md`)
- Extends "how to contribute" significantly. (`docs/contributing.md`)
- Adds a PR template with a checklist. This template is activated automatically based on its name and location (in `.github`).

I also noticed we had two instances of `contributing.md`, so made sure the one in the top directory points to the one in RTD.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -1,24 +1,109 @@
...
+
+7. Make sure the tests pass by running the following command from the top of
+   the repository:
+
+   ```bash
+   ./tests/run_all_tests.sh
+   ```
+
+8. Once your change is done, create a commit as follows (
+   [how to write a commit message](https://chris.beams.io/posts/git-commit/)):
```
Comment (avital) [inline docs/contributing.md:81]: This is nice

File: docs/contributing.md [hunk 2]
```diff
@@ -1,24 +1,109 @@
...
+Everyone can contribute to Flax, and we value everyone's contributions. 
+You can contribute in many more ways than just writing code. Anwering questions
+on our [Discussions page](https://github.com/google/flax/discussions), helping
+each other, and improving our documentation are extremely valuable to our
+ecosystem.
 
-## Getting started
+We also appreciate if you spread the word, for instance by starring our Github
+repo, referencing Flax in blog posts of projects that used it, or giving us a
+Twitter shoutout by using the hashtag "#flax".
```
[the remark this answers is not in the document]
Comment (avital) [reply docs/contributing.md:11]: Also, Is #flax used for other things now? Flax is a common term outside of ML...

**Review by avital — APPROVED** [review]
This is a massive improvement, thanks Marc!

I added some small comments.

File: docs/contributing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcvanzee) [reply docs/contributing.md:11]: I agree we should have our own Twitter account, and this tag was my idea of having an alternative to that. I'll remove the Twitter part here.

File: docs/contributing.md (hunk 1 above)
Comment (marcvanzee) [reply docs/contributing.md:81]: I agree!

**Pull request merged**